elasticsearch/templates/service.yaml

```
@@ -18,13 +18,13 @@ spec:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
     app: "{{ template "uname" . }}"
   ports:
-  - name: http
+  - name: {{ .Values.service.httpPortName | default "http" }}
```

The defaults can be removed from here now that you have added the defaults into the values.yaml.

```suggestion
  - name: {{ .Values.service.httpPortName }}
```